### DISCUSSION TITLE
Code that controls power button action

### DISCUSSION THREAD

**Discussion by 0xDEAD-C0DE** [YaLTeR/niri#158] [Q&A] [answered]
Good afternoon. Since, I believe this is not an issue with Niri, I decided to post it here. 
The config states that Niri handles the action of the power button when pressed. I do not start Niri from a niri.session ( Login Graphical Interface). I simply execute `niri` from TTY1` 
Where is the Rust code that handles this action located? I want to inspect it for learning it purposes .

When I press the power button, the computer powers off and when I power it back on it sends me to the Dell diagnostic test. 
I am Archlinux `kernel 6.7.2 x86_64` DELL XPS 15 9550` `systemd 255.3.1`

**Comment by YaLTeR** [comment, marked as answer]
Hey, so when you run niri directly, it does not do any power button handling. In this case it's controlled by logind or something like that. So in your case you need to look into logind.conf.

When niri is a session however, it will:
1. Inhibit the logind power key handling, so that niri can handle it by itself: https://github.com/YaLTeR/niri/blob/main/src/main.rs#L246, https://github.com/YaLTeR/niri/blob/main/src/niri.rs#L1042
2. When it's pressed, ask logind to suspend: https://github.com/YaLTeR/niri/blob/main/src/backend/tty.rs#L1511

**Reply by 0xDEAD-C0DE** [reply]
Thank you for your response. Yes, I am aware of the logind.conf for systemd.
